activemodel/lib/active_model/attribute_registration.rb

```diff
@@ -72,6 +73,10 @@ def resolve_attribute_name(name)
         def resolve_type_name(name, **options)
           Type.lookup(name, **options)
         end
+
+        def define_predicate_method(name)
+          define_method("#{name}?") { public_send(name) == true }
```

Comment:
Just to clarify, `== true` part was added intentionally to make the method strictly return a boolean? (Never `nil`)

I think this is a reasonable design even though technically we have an option to drop this comparison as most of the values will be coerced to `true` or `false` and `nil` is falsey so it wouldn't be wrong. 

But personally I'm leaning towards keeping the comparison to avoid situations like:
`Person.new.loves_ruby? # => nil` which from falseness perspective is "No" but in reality could mean "the person is not sure yet"

Reply:
@nvasilevski Can you think of a situation where we make this distinction? I was under the assumption that we care more about the [semantics](https://edgeguides.rubyonrails.org/api_documentation_guidelines.html#booleans) than the specific values 🤔